### ISSUE TITLE
The tests should automatically be called for all available platforms

### ISSUE THREAD

**Issue by KillaPL** [ruby2d/ruby2d#47]
Calling `rake` should run the rspec tests for all available platforms. Just wanted to put it here, and invite you for discussion : )

**Comment by blacktm** [comment]
By platforms, do you mean MRuby and Opal? If so, that's a challenge given RSpec was built for and only supports MRI. I see there are ports people are working on, like [mruby-spec](https://github.com/sagmor/mruby-spec) and [opal-rspec](https://github.com/opal/opal-rspec).

**Comment by KillaPL** [comment]
Yeah, I mean opal and mruby. Well, it's something worth considering, currently it's really easy to write code that runs on mri nicely, but fails on either mruby or opal : (

**Comment by blacktm** [comment]
Well, one idea is to drop RSpec entirely and roll our own custom thing that works everywhere. Might sound painful, but it's probably reasonable. This isn't some complicated Rails app or anything — we're just trying to run code and assert some values. For Simple 2D, I looked at a bunch of test frameworks for C before concluding that it would be easier to write a little test suite from scratch, [see here](https://github.com/simple2d/simple2d/blob/master/test/auto.c). Writing a generalized test runner is very hard, but writing one narrowly focused on your own library is totally feasible.

On the other hand, MRuby and Opal are pretty much on par with MRI interface wise (excluding a few methods, but maybe you've had a different experience), and the more we move out of the native extensions and into Ruby land, the better we'll be. In that case, running the automated tests in MRI might be good enough, I don't know.

**Comment by KillaPL** [comment]
Native rake/test from ruby seems to be pretty simple, might be worth checking out

**Comment by blacktm** [comment]
Just cleaning up stale issues. Testing is much better in 9650c7b3c9811bd17d2c6e6205d5660cc9dbf788!
